packages/clerk-js/src/ui/hooks/useFetchRoles.ts

```diff
@@ -14,9 +15,15 @@ const getRolesParams = {
 };
 export const useFetchRoles = (enabled = true) => {
   const { organization } = useOrganization();
+  const canManageMemberships = useProtect({ permission: 'org:sys_memberships:manage' });
+  const canReadMemberships = useProtect({ permission: 'org:sys_memberships:read' });
+
   // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
   const getRoles = ({ pageSize, initialPage }: GetRolesParams) => organization!.getRoles({ pageSize, initialPage });
-  const { data, isLoading } = useFetch(enabled && !!organization?.id ? getRoles : undefined, {
+
+  const hasSystemPermissions = canManageMemberships || canReadMemberships;
```

Comment:
We have this hook being called from `OrganizationProfile` pieces and `OrganizationSwitcher`, so instead of applying this condition in all places, I believe it should stay in the query itself.

Reply:
Technically, https://github.com/clerk/javascript/pull/6703/files#r2320397829 would already solve the issue of the 403 on the roles query when rendering `OrganizationSwitcher`, however I think it's worth to place an extra layer of protection here for future usage in other org components.